app/jobs/hyku_addons/reindex_model_job.rb

```diff
@@ -21,18 +19,29 @@ def perform(klass, cname, limit: 25, page: 1, options: {})
       @page = page
       @klass = klass
       AccountElevator.switch!(cname)
-      offset = (page - 1) * @limit
-      work_class = klass.constantize
+      @offset = (page - 1) * @limit
 
       if @use_work_ids.present?
-        fetch_work_using_ids(work_class)
+        fetch_work_using_ids
       else
-        reindex_and_mint_work(work_class, offset)
+        reindex_and_mint_work
       end
     end
 
     private
 
+      def works
```

Comment:
These changes clean up the class a little and allow the new code to share logic with the existing code.  However they are not strictly needed for this update.  Only the `total_entries` method is needed and that could be refactored if you do not wish to add all these refactors.